docs/validate_customizations.md

```diff
@@ -0,0 +1,172 @@
+		* Permission changes often need to be immediate for business operations
+	- Instead, manage permissions directly through the UI's Role Permission Manager
+
+2. **Module Attribution** (`validate_module`)
+	- Verifies that all Custom Fields and Property Setters have a valid module key
+	- Ensures Property Setters are associated with the correct module and application
+
+3. **System Generation Status** (`validate_system_generated`)
+	- Identifies and flags system-generated Custom Fields that may be managed by internal Frappe/ERPNext APIs
+		- Examples include barcode visibility in Stock Settings
+		- The management of Accounting Dimension or Inventory Dimensions fields can be overridden if subsequent customization is required 
+
+4. **Own DocType Customization** (`validate_customizations_on_own_doctypes`)
+	- Prevents customizing DocTypes that are already defined within the same app, ensuring modifications to an app's DocTypes are made in the source rather than through customizations
+
+5. **Duplicate Customizations** (`validate_duplicate_customizations`)
+	- Detects duplicate Custom Fields  and Property Setters across different modules
+	- Prevents conflicting customizations between apps
+
+## Adding New Customizations
+
+Customizations can be added to the system through several methods:
+
+1. **Via Customize Form**
+	- Navigate to the "Customize Form" page
+	- Make desired modifications
+	- Use the "Export Customizations" button to save all changes
+		- This may include customizations made by ERPNext, HRMS or other apps, so editing the resulting JSON file may be appropriate
+		- This is not usually the best choice for adding a new Custom Field or Property Setter to an existing customization file
+
+2. **Individual Field/Property Export**
+	- Access the Custom Field or Property Setter list
+	- Use the 'Copy to Clipboard' option to collect a JSON serialization of the record
+	- Paste into your app's customization JSON file in the appropriate section
+
+3. **Programmatic Export**
+	- Write scripts to export customizations
+	- Ensure proper module attribution
+	- Place files in the correct app directory structure
+
+## File Structure
+
+Customizations should be organized in the following directory structure so that they are loaded automatically by Frappe.
+```
+your_app/
+├── your_app/
+│   ├── modules.txt
+│   └── module_name/
+│       └── custom/
+│           └── name_of_customized_doctype.json
+```
+
+## Why To Not Use Fixtures
+
+While Frappe provides a fixtures feature that can export and import DocTypes including Custom Fields and Property Setters, it has several drawbacks that make it unsuitable for managing customizations:
+
+1. **No Separation of Concerns**
+  - Fixtures don't maintain any reference to their source app; all Custom Fields and Property Setters are included
+  - Fixtures are organized into a file for Custom Fields one for Property Setters, without any organization by doctype or module.

```

Comment:
"...Fields one for Property Setters..." -> "...Fields and for Property Setters..."